core/src/main/java/nl/nn/adapterframework/jms/PullingJmsListener.java

```diff
@@ -212,12 +211,12 @@ public void afterMessageProcessed(PipeLineResult plr, Object rawMessageOrWrapper
 				if (session==null) { 
 					try {
 						session=getSession(threadContext);
-						send(session, replyTo, cid, prepareReply(plr.getResult(),threadContext), getReplyMessageType(), timeToLive, stringToDeliveryMode(getReplyDeliveryMode()), getReplyPriority(), ignoreInvalidDestinationException);
+						send(session, replyTo, cid, prepareReply(plr.getResult(),threadContext).asString(), getReplyMessageType(), timeToLive, stringToDeliveryMode(getReplyDeliveryMode()), getReplyPriority(), ignoreInvalidDestinationException);
 					} finally {
 						releaseSession(session);					 
 					}
 				}  else {
-					send(session, replyTo, cid, plr.getResult(), getReplyMessageType(), timeToLive, stringToDeliveryMode(getReplyDeliveryMode()), getReplyPriority(), ignoreInvalidDestinationException); 
+					send(session, replyTo, cid, plr.getResult().asString(), getReplyMessageType(), timeToLive, stringToDeliveryMode(getReplyDeliveryMode()), getReplyPriority(), ignoreInvalidDestinationException); 
 				}
 			} else {
```

Comment:
Ik denk dat het mooier is als de send message hier een adaperframework message accepteert. Wij gebruiken nu alleen jms TextMessages, maar er zijn er nog meer.. (BytesMessage, MapMessage, ObjectMessage, StreamMessage).
Die hoeft niet nu, maar misschien handig als comment ergens bij te zetten.